subgraphs/_reference_/src/common/tokens.ts

```diff
@@ -0,0 +1,184 @@
+/* eslint-disable prefer-const */
+import { ERC20 } from "../../generated/UniswapV2Factory/ERC20";
+import { ERC20SymbolBytes } from "../../generated/UniswapV2Factory/ERC20SymbolBytes";
+import { ERC20NameBytes } from "../../generated/UniswapV2Factory/ERC20NameBytes";
+import { Address } from "@graphprotocol/graph-ts";
+
+export const INVALID_TOKEN_DECIMALS = 9999;
+export const UNKNOWN_TOKEN_VALUE = "unknown";
+
+export function fetchTokenSymbol(tokenAddress: Address): string {
+  let contract = ERC20.bind(tokenAddress);
+  let contractSymbolBytes = ERC20SymbolBytes.bind(tokenAddress);
+
+  // try types string and bytes32 for symbol
+  let symbolValue = UNKNOWN_TOKEN_VALUE;
+  let symbolResult = contract.try_symbol();
+
+  // non-standard ERC20 implementation
+  if (symbolResult.reverted) {
```

Comment:
how about

```
if (!symbolResult.reverted) {
  symbolValue = symbolResult.value;
  return;
}
...
```

To make it less nested?

Reply:
good idea!